examples/tests/self-managed-ng-os/index.ts

```diff
@@ -0,0 +1,70 @@
+import * as awsx from "@pulumi/awsx";
+import * as aws from "@pulumi/aws";
+import * as eks from "@pulumi/eks";
+import * as iam from "./iam";
+
+// IAM roles for the node groups.
+const role = iam.createRole("self-managed-ng-os");
+const instanceProfile = new aws.iam.InstanceProfile("self-managed-ng-os", {role: role});
+
+// Create a new VPC
+const eksVpc = new awsx.ec2.Vpc("self-managed-ng-os", {
+  enableDnsHostnames: true,
+  cidrBlock: "10.0.0.0/16",
+});
+
+// Create an EKS cluster.
+const cluster = new eks.Cluster("self-managed-ng-os", {
```

Comment:
This is going to generate a custom userdata implicitly somewhere? Trying to correlate the example to the goal of the PR.

Reply:
Yes, self managed node groups need user data because they deploy an auto scaling group for the nodes. Everything (lifecycle, config, etc.) is self managed and needs to be provided for those to work. So this requires the generated user data.

With Managed Node Groups, there's a base configuration AWS provides and everything users configure gets merged with the base configuration. In most cases this doesn't result in custom user data. But for certain configuration parameters it is necessary.